polars/polars-core/src/chunked_array/kernels/take.rs

```diff
@@ -448,14 +442,12 @@ pub(crate) unsafe fn take_utf8(
         return Arc::new(builder.finish());
     }
 
-    let mut data = ArrayData::builder(ArrowDataType::LargeUtf8)
-        .len(data_len)
-        .add_buffer(offset_buf.into())
-        .add_buffer(values_buf.into_arrow_buffer());
-    if let Some(null_buffer) = nulls {
-        data = data.null_bit_buffer(null_buffer);
-    }
-    Arc::new(LargeStringArray::from(data.build()))
+    // Safety: all "values" are &str, and thus valid utf8
+    Arc::new(Utf8Array::<i64>::from_data_unchecked(
+        offset_buf.into(),
+        values_buf.into(),
+        validity,
+    ))
```

Comment:
Arrow2 has no "ArrayData" or its builder: each concrete array type has its own constructor that expects the corresponding buffers / bitmaps / childs. In this case, `Utf8Array` requires 3 items.

Also, note that `from_data_unchecked` skips the utf8 checks: every array in arrow2 checks that their buffers respect the arrow specification, and `panic!` if the buffers do not fulfill it. In this case, all offsets much be monotonically increasing, and the last offsets' value must equal the length of the values buffer.

`values_buf.into()` because a `MutableBuffer<T>` implements `Into<Buffer<T>>` and `MutableBitmap` implements `Into<Option<Bitmap>>`.

Reply:
> every array in arrow2 checks that their buffers respect the arrow specification, and panic! if the buffers do not fulfill it. In this case, all offsets must be 

Would you consider an `unsafe` version as well? That does not do the monotonically increasing check. In a correctly written `take` kernel (which I want to be as fast as possible :) ), this check seems redundant.

Reply:
I have not checked performance, but we can consider that. The reason I was a bit picky is that when these arrays are used in IPC or FFI, we "leak" inconsistent arrays around, which causes problems beyond Rust alone.